Data: https://drive.google.com/drive/folders/1ZYLRLzx3iBiG9GCfV9UtFBFZWkzvF7El?usp=sharing

Homework questions:

Q1. Which SputnikDAO has approved the most grants?
Q2. Which SputnikDAO has distributed the most NEAR" via grants?
Q3. Who voted to approve the grant for “Budget for NEAR Arabia Guild” in the Marketing Vertical guild?
Q4. How many “Yes” votes did nft-gaming.sputnikdao.near’s proposal to the CommunitySquadDAO receive?
Q5. How much $NEAR was requested from the Creatives DAO by chloe.near for TipBot funding?

Reflection questions:
Descripe the tools & process you used:

Any challenges / difficulties (i.e. dataset issues, definitions):

Is there any raw data missing that might be interesting to analyze?

Which datapoints do you think are relevant to measuring system health?

In [1]:
import pandas as pd
import numpy as np

# Q1. Which SputnikDAO has approved the most grants?

In [3]:
fdir = 'data/'
fname = 'Sputnik DAO Stats.xlsx'
df_daostats = pd.read_excel(fdir+fname,usecols="A:H")


In [10]:
df_daostats.head()

Name  Ⓝ Value  Value (USD)  Proposals  In Progress  \
0     genesis.sputnikdao.near    15073       157965         45            0   
1     academy.sputnikdao.near       48          503          5            0   
2   ethdenver.sputnikdao.near      264         2767        133            0   
3        arts.sputnikdao.near      205         2144          8            0   
4  createbase.sputnikdao.near      503         5272        483            4   

   Successful  Failed  Expired  
0          15       9       21  
1           2       0        3  
2          92      12       28  
3           7       0        1  
4         370      77       22

In [14]:
df_daostats.sort_values(by="Successful",).iloc[-3:]

Name  Ⓝ Value  Value (USD)  Proposals  \
23  communitysquad.sputnikdao.near     3494        36622        175   
4       createbase.sputnikdao.near      503         5272        483   
31         sandbox.sputnikdao.near     8778        91993        455   

    In Progress  Successful  Failed  Expired  
23            4         144       3        9  
4             4         370      77       22  
31            1         388       6       26

**Conclusion: Sandbox has most sucessful proposals (388), followed by createbase and communitysquad**

# Q2. Which SputnikDAO has distributed the most ''NEAR" via grants?

In [22]:
fname = 'Sputnik DAO Proposals.xlsx'
df_dao_proposals = pd.read_excel(fdir+fname,sheet_name=None)
df_grants={'dao':[],'grants':[]}
for daoname in df_dao_proposals.keys():
    if daoname=='WORKSPACE': # this is not a dao, so skip it
        continue
    df = df_dao_proposals[daoname]
    df.columns = df.columns.str.replace(' ', '_').str.lower() # clean column names to avoid pandas key error
    if 'amount_(near)' in df.keys():
        # note: some numbers are just "x" so we have to skip them
        # only including proposals that succeeded
        totalfund = pd.to_numeric(df['amount_(near)'].loc[df['status']=='Success'],errors = 'coerce').sum(skipna=True)
    else:
        
        totalfund = (pd.to_numeric(df['kind__amount'].loc[df['status']=='Success'],errors = 'coerce')/(10**24)).sum(skipna=True)
    df_grants['dao'].append(daoname)
    df_grants['grants'].append(totalfund)
df_grants = pd.DataFrame(df_grants).sort_values(by='grants')
    

In [23]:
df_grants.iloc[-3:]

dao       grants
5  Ecosystem (KoG)  101100.0000
3      Human Guild  163109.0100
0  Community Squad  278390.0073

**Conclusion: Community Squad distributed most grant!**

# Q5. How much NEAR was requested from the Creatives DAO by chloe.near for TipBot funding?

In [24]:
df_dao_proposals.keys()

dict_keys(['WORKSPACE', 'Community Squad', 'Creatives', 'Marketing Vertical', 'Human Guild', 'Open Web Sandbox', 'Ecosystem (KoG)', 'Degens', 'Assembly Script', 'Genesis', 'Move Capital', 'NxM', 'Bounties', 'Grants', 'Createbase'])

In [33]:
df = df_dao_proposals['Creatives']
totalnear = 0
for k,row in df.loc[df['proposer']=='chloe.near'].iterrows(): # all the proposals from chloe.near
    if ('tip' in row['description'].lower()) and row['status']=='Success':
        totalnear += row['amount_(near)']
print(totalnear)        

150.0


In [32]:
df.loc[df['proposer']=='chloe.near'] # directly check -- correct!

status    proposer       target_(recipient) amount_(near)  \
2   Success  chloe.near               chloe.near             x   
27  Success  chloe.near               chloe.near          50.0   
44  Success  chloe.near               chloe.near          50.0   
56  Success  chloe.near               chloe.near          50.0   
73  Success  chloe.near  marmaj.sputnik-dao.near         190.0   

                                          description  \
2                   Updating purpose of creatives dao   
27              50 N for tip bot with creatives on TG   
44  August tip bot payout to support Creative Comm...   
56  Tip bot budget for chloe to incentivize commun...   
73  $1500 USD (NEAR @ 7.9) = 190 NEAR for NFT coll...   

                                     forum_discussion      vote_period_end  \
2   https://gov.near.org/t/guide-the-creatives-dao...  1625062749500654800   
27  https://gov.near.org/t/budget-creatives-dao-bu...  1629928990803414300   
44  https://gov.near.org/t/budget-creatives-dao-bu...  1632448532535943200   
56  https://gov.near.org/t/budget-creatives-dao-bu...  1635031363712235500   
73  https://gov.near.org/t/approved-marmaj-dao-oct...  1636061194091931000   

    vote_yes  vote_no rebecca.near tabear.near jmm.near vandal.near chloe.near  
2        2.0      0.0          Yes         NaN      NaN         NaN        Yes  
27       2.0      0.0          Yes         NaN      NaN         NaN        Yes  
44       2.0      0.0          Yes         NaN      NaN         NaN        Yes  
56       2.0      0.0          Yes         NaN      NaN         NaN        Yes  
73       3.0      0.0          NaN         Yes      Yes         Yes        NaN

**Conclusion: $150 NEAR**